src/runtime/components/Button.vue

```diff
@@ -35,9 +35,9 @@ export interface ButtonProps extends UseComponentIconsProps, Omit<LinkProps, 'ra
 }
 
 export interface ButtonSlots {
-  leading(props?: {}): any
+  leading(props?: { ui: Button['slots'] }): any
   default(props?: {}): any
-  trailing(props?: {}): any
+  trailing(props?: { ui: Button['slots'] }): any
```

Comment:
I think you need this to have correct types:
```suggestion
  leading(props: { ui: { [K in keyof Required<Button['slots']>]: (props?: Record<string, any>) => string } }): any
  default(props?: {}): any
  trailing(props: { ui: { [K in keyof Required<Button['slots']>]: (props?: Record<string, any>) => string } }): any
```